geometric_kernels/lab_extras/extras.py

```diff
@@ -11,6 +11,26 @@
 def take_along_axis(a: B.Numeric, index: B.Numeric, axis: int = 0):
     """
     Gathers elements of `a` along `axis` at `index` locations.
+
+    The index array will be flattened.
+
+    :param a: [N1, ..., Nj, ..., Nn]
+    :param index: [J...]
+
+    :return: [N1, ..., J, ..., Nn]
+    """
+
+
+@dispatch
+@abstract()
+def take_along_last_axis(a: B.Numeric, indices: B.Numeric) -> B.Numeric:
+    """
+    Takes elements of `a` along the last axis. `indices` must be the same rank (ndim) as
+    `a`. Useful in e.g. argsorting and then recovering the sorted array.
+    ```
+
+    E.g. for 3d case:
+    output[i, j, k] = a[i, j, indices[i, j, k]]
```

Comment:
Why does `indices` need to have the same rank as `a`? Would this not work?
```suggestion
    output[i, j, k] = a[i, j, indices[i, j]]
```

Reply:
This solution is the only one I found that works for all `np`, `tf`, `torch` and `jax`. E.g `torch.gather` requires indices be the same rank as the input array, while `tf.gather` uses a completely different `indices` structure. However, when indices are the same rank as the input, **and** the elements are taken along the last axis, this works with every framework.

Prob should write a unit-test to double-check that.